### PR TITLE
Add CPU parts of the performance-related hyperparameters logic

### PR DESCRIPTION
**Pull request by Vika-F**
Changes in this PR:

- Base algorithm::Hyperparameter class implemented in DAAL.
- Pointer to the algorithm::Hyperparameter, _hpar,  was added as the protected, constant, field of the base Algorithm and AlgorithmContainer classes
- Passing the Hyperparameters from DAAL interface into DAAL kernel implemented by analogy with DAAL's Parameter object
- Initial version of Hyperparameter class for DAAL Covariance algorithm implemented
- Passing the hyperparameters from oneDAL to DAAL is implemented in Covariance and PCA + Covariance algorithms. For the PCA only the hyperparameters for the Covariance part of the algorithm are passed into DAAL for now.

What remains to be done:

- Move the temporary logic of block size calculation that exists now in both Covariance and PCA algorithms on oneDAL side to some common place. I.e. get rid of copy/paste.
- Add hyperparameters passing into other DAAL algorithms' kernels that are called from oneDAL API to be able to pass hyperparameters from oneDAL further. Now the hyperparameters can be passed only into Covariance algorithm.

### CODE REVIEW

File: cpp/oneapi/dal/algo/covariance/test/fixture.hpp [hunk 1]
```diff
@@ -136,7 +136,6 @@ class covariance_test : public te::crtp_algo_fixture<TestType, Derived> {
 
     void online_general_checks(const te::dataframe& input, const te::table_id& input_table_id) {
         const table data = input.get_table(this->get_policy(), input_table_id);
-        const std::int64_t nBlocks = 10;
```
[the remark this answers is not in the document]
Comment (md-shafiul-alam) [reply cpp/oneapi/dal/algo/covariance/test/fixture.hpp:139]: I think the idea of this PR is not to set hyperparameters directly in the code.

[the remark this answers is not in the document]
Comment (ethanglaser) [reply cpp/oneapi/dal/algo/covariance/test/fixture.hpp:139]: Hmm, any idea how its supposed to be set? Its needed here: https://github.com/oneapi-src/oneDAL/blob/master/cpp/oneapi/dal/algo/covariance/test/fixture.hpp#L147-L148, causing CI failures without definition

[the remark this answers is not in the document]
Comment (md-shafiul-alam) [reply cpp/oneapi/dal/algo/covariance/test/fixture.hpp:139]: Yes, it should be defined here and other examples. on it.

**Review by ethanglaser — APPROVED** [review]
CI looks good, maybe would want an approval from Nikita or someone else in terms of implementation details

**Review by KulikovNikita — APPROVED** [review]

**Pull request merged**